ethereum/contracts/Messages.sol

```diff
@@ -24,25 +24,22 @@ contract Messages is Getters {
     *  - it aims to ensure the guardianSet is not expired
     *  - it aims to ensure the VM has reached quorum
     *  - it aims to verify the signatures provided against the guardianSet
+    *  - it aims to verify the hash field provided against the contents of the vm
     */
     function verifyVM(Structs.VM memory vm) public view returns (bool valid, string memory reason) {
         (valid, reason) = verifyVMInternal(vm, true);    
     }
 
     /**
     * @dev `verifyVMInternal` serves to validate an arbitrary vm against a valid Guardian set
-    *  - it is similar to verifyVM but drops the verification of the hash as it is called on the result of parseVM whose vm.hash field can be trusted.
-    *  - it aims to make sure the VM is for a known guardianSet
-    *  - it aims to ensure the guardianSet is not expired
-    *  - it aims to ensure the VM has reached quorum
-    *  - it aims to verify the signatures provided against the guardianSet
+    * if checkHash is set then the hash field of the vm is verified against the hash of its contents
```

Comment:
maybe explain here why this is not something we always want to do? adding a comment in parseAndVerifyVM where we call this function with false explaining why would also be nice

Reply:
makes sense, added